src/bank.rs

```diff
@@ -4,27 +4,78 @@
 //! already been signed and verified.
 
 use bincode::{deserialize, serialize};
+use budget;
 use chrono::prelude::*;
 use counter::Counter;
 use entry::Entry;
 use hash::{hash, Hash};
+use instruction::Instruction;
 use itertools::Itertools;
 use ledger::Block;
 use log::Level;
 use mint::Mint;
-use payment_plan::{Payment, PaymentPlan, Witness};
+use payment_plan::{Payment, PaymentPlan};
 use signature::{Keypair, Pubkey, Signature};
 use std;
-use std::collections::hash_map::Entry::Occupied;
 use std::collections::{BTreeMap, HashMap, HashSet, VecDeque};
 use std::result;
 use std::sync::atomic::{AtomicUsize, Ordering};
 use std::sync::RwLock;
 use std::time::Instant;
 use timing::{duration_as_us, timestamp};
-use transaction::{Instruction, Plan, Transaction};
+use transaction::Transaction;
 use window::WINDOW_SIZE;
 
+/// An Account with userdata that is stored on chain
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+pub struct Account {
+    /// tokens in the account
+    pub tokens: i64,
```

Comment:
why do we permit negative tokens?

Reply:
@mvines @garious had the idea that we could have validators vote with a negative fee.  so we are supporting the possibility of negative tokens.

Reply:
Yeah, I still like to believe we'll be able to coalesce transactions from the right to trivialize rollback. I hijacked that top bit to keep the dream alive.